src/lib/holocene/modal.svelte

```diff
@@ -21,45 +26,82 @@
     dispatch('cancelModal');
   };
 
-  const handleKeyUp = (event: KeyboardEvent) => {
+  const handleKeyboardNavigation = (event: KeyboardEvent) => {
+    if (!open) {
+      return;
+    }
+
     if (event.key === 'Escape') {
       cancelModal();
+      return;
+    }
+
+    if (event.key === 'Tab') {
+      if (event.shiftKey) {
+        if (document.activeElement === closeButton) {
+          confirmButton.focus();
+          event.preventDefault();
+        }
+      } else if (document.activeElement === confirmButton.buttonElement) {
+        closeButton.focus();
```

Comment:
There's an edge case here where if `loading` is true the focus can escape the modal. 

What do you think about a more generic check for a first and last focusable element? e.g.

```js
    const focusable = modalElement.querySelectorAll('button');
    const firstFocusable = focusable[0];
    const lastFocusable = focusable[focusable.length - 1];
    if (event.key === 'Tab') {
      if (event.shiftKey) {
        if (document.activeElement === firstFocusable) {
          lastFocusable.focus();
          event.preventDefault();
        }
      } else if (document.activeElement === lastFocusable) {
        firstFocusable.focus();
        event.preventDefault();
      }
    }
```

This means we could also remove the `focus` export on the `Button` component.

Reply:
Ah nice catch. Yeah I had initially thought of a more generic solution like that, but I didn't love the idea of using a `querySelectorAll` to get all the focusable elements, I thought using `bind` with only the elements we wanted focusable was safer. That said, I don't know why we wouldn't want any button in a modal to _not_ be focusable, so I think I like this solution better 👍